tests/CMakeLists.txt

```diff
@@ -21,12 +22,14 @@ function(define_test testfile np backend)
 endfunction()
 
 define_test(test_allocator.f90 1 omp)
+define_test(test_reordering.f90 1 omp)
 define_test(omp/test_omp_tridiag.f90 ${CMAKE_CTEST_NPROCS} omp)
 define_test(omp/test_omp_transeq.f90 ${CMAKE_CTEST_NPROCS} omp)
 define_test(omp/test_omp_dist_transeq.f90 ${CMAKE_CTEST_NPROCS} omp)
 
 if(${CMAKE_Fortran_COMPILER_ID} STREQUAL "PGI" OR
     ${CMAKE_Fortran_COMPILER_ID} STREQUAL "NVHPC")
+  define_test(test_reordering.f90 1 cuda)
```

Comment:
Please see my comment on the test program. I think defining the test like this with the current setup only repeats the test with the CPU kernels. We need to pass a `-DCUDA` flag to the compiler so that the correct classes are instantiated. But because the test program is not ready as of now passing the `-DCUDA` flag will result in errors. Once the test program is able to test the CUDA backend we can have this line back so that with NVIDIA compiler we test both CPU and GPU kernels.